File: 2024/javascript/06-isTheGiftInsideTheBox.js
// 06. Is The Gift Inside The Box?

/** @param {string[]} box
 *  @returns {boolean} True if the gift is inside the box
 */
function inBox(box) {
  box = box.slice(1, -1);

  for (const line of box) {
    if (
      line.includes("*") &&
      line.indexOf("*") > 0 &&
      line.indexOf("*") < line.length - 1
    ) {
      return true;
    }
  }

  return false;
}

console.log(inBox(["###", "#*#", "###"])); // ➞ true
console.log(inBox(["####", "#* #", "#  #", "####"])); // ➞ true
console.log(inBox(["#####", "#   #", "#  #*", "#####"])); // ➞ false
console.log(inBox(["#####", "#   #", "#   #", "#   #", "#####"])); // ➞ false

File: 2024/python/06_is_the_gift_inside_the_box.py
"""Challenge #6: 📦 Is The Gift Inside The Box?"""


def in_box(box: list[str]) -> bool:
    """
    In Box.

    Args:
        box (list[str]): The box to check

    Returns:
        bool: True if the gift is inside the box
    """
    box = box[1:-1]

    for line in box:
        if "*" in line and line.index("*") > 0 and line.index("*") < len(line) - 1:
            return True

    return False


print(
    in_box(
        [
            "###",
            "#*#",
            "###",
        ]
    )
)  # True

print(
    in_box(
        [
            "####",
            "#*#",
            "##",
            "####",
        ]
    )
)  # True

print(
    in_box(
        [
            "#####",
            "##",
            "##*",
            "#####",
        ]
    )
)  # False

print(
    in_box(
        [
            "#####",
            "#   #",
            "#   #",
            "#   #",
            "#####",
        ]
    )
)  # False
